pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -567,34 +565,10 @@ func (p *criStatsProvider) cleanupOutdatedCaches() {
 // removeTerminatedContainer returns the specified container but with
 // the stats of the terminated containers removed.
 func removeTerminatedContainer(containers []*runtimeapi.Container) []*runtimeapi.Container {
-	containerMap := make(map[containerID][]*runtimeapi.Container)
-	// Sort order by create time
-	sort.Slice(containers, func(i, j int) bool {
-		return containers[i].CreatedAt < containers[j].CreatedAt
-	})
-	for _, container := range containers {
-		refID := containerID{
-			podRef:        buildPodRef(container.Labels),
-			containerName: kubetypes.GetContainerName(container.Labels),
-		}
-		containerMap[refID] = append(containerMap[refID], container)
-	}
-
 	result := make([]*runtimeapi.Container, 0)
-	for _, refs := range containerMap {
-		if len(refs) == 1 {
-			result = append(result, refs[0])
-			continue
-		}
-		found := false
-		for i := 0; i < len(refs); i++ {
-			if refs[i].State == runtimeapi.ContainerState_CONTAINER_RUNNING {
-				found = true
-				result = append(result, refs[i])
```

Comment:
After I revisited this code, I found that it did not actually perform any deduplication. I think it is indeed useless.
Suppose there are indeed multiple running containers with the same name. Since we didn't break the for loop when we set found to true, we will continue to append other running containers with the same name to result.

Reply:
This function does two things, as follows :
- Find all running containers created by k8s
- If the containers of one pod are all in `non-runing` state, find the last created container.

Reply:
To be precise, the premise is that we no longer need containers that are not running. For the first thing, all we need to do is traverse the containers array to pick the running containers.

Reply:
See: https://github.com/kubernetes/kubernetes/issues/53514#issuecomment-334629943

```go
 // A ContainerInfo is considered to be of a terminated container if it has an 
 // older CreationTime and zero CPU instantaneous and memory RSS usage.
```
This only exclude duplicated terminated containers(**duplicated** and **terminated**), not **terminated**.

Reply:
If you are willing to look at the problem that the PR #74336 is trying to solve, I think you will understand what I mean.

TL;DR: These terminated container stats will make the metrics-server not work as expected, now I want to modify this code to filter out the terminated containers.

Reply:
I know what you mean, but if filter out all terminated containers, that may cause other problems.  
/cc @Random-Liu @yguo0905

Reply:
I do not think so. Taking cadvisor as an example, except for the uncleaned cgroup directory problem #47853, the containers are no longer tracked by cadvisor once they are terminated. After the terminated container is filtered out, the result returned by cri and cadvisor is equivalent. 

Please let me know if my understanding is not correct.